pkg/operator/ceph/controller/predicate.go

```diff
@@ -543,16 +554,28 @@ func isValidEvent(patch []byte, objectName string) bool {
 	// Re-marshal to get the last diff
 	patch, err = json.Marshal(p)
 	if err != nil {
-		logger.Errorf("failed to marshal patch. %v", err)
+		logErrorUnlessSensitive("failed to marshal patch", err, patchContainsSensitiveData)
 		return false
 	}
```

Comment:
json.Marshal (and json.Unmarshal) don't output a lot of log info outside of the specific error (like +Inf not supported or unexpected end of JSON input), but in the interest of safety, I think it's best to still redact the error just in case it outputs any of the content of the (potentially malformed) JSON blob that might contain secret info. But let me know what you think.

Reply:
it does seem unlikely that the json marhal or unmarshal would print an error with the secret contents, but better safe than sorry